enhancements/console/pinned-resources.md

```diff
@@ -0,0 +1,190 @@
+
+Some customers want to provide defaults for those pre-pinned resources for all users.
+
+On the Developer perspective navigation, users can see the default pinned resources if they have not customized these pinned resources. Cluster admins have no way to set these default pinned resources for their users. Users can change their pinned resources on their own. The user-pinned resources are saved in the user settings (as-per user-settings ConfigMap).
+
+This enhancement allows cluster admins to define pre-pinned resources for new users and users who have not customized their navigation items. Users should still be able to change these defaults and customize the pinned resources on their own. New changes from a cluster admin after a user customize their own pinned resources do not have to be shown.
+
+![Default pinned resources](default-pinned-resources.png)
+
+## Motivation
+
+Some customers want to provide default pre-pinned resources for all users.
+
+### User Stories
+
+#### Story 1
+
+Customer wants to customize the developer perspective navigation by providing pre-pinned resources to users who do not have their navigation items customized.
+
+As a cluster admin, I want to provide default pre-pinned resources for all users so that the developers who donot have their navigation items customized will see these default pinned resources.
+
+### Goals
+
+- Allow cluster admins to define default pre-pinned resources on the dev perspective navigation
+
+### Non-Goals
+
+N/A
+
+## Proposal
+
+We can extend the existing `operator.openshift.io/v1` / `Console CRD`.
+We can add `spec.customization.perspectives.pinnedResources` which lists default pre-pinned resources
+
+### Workflow Description
+
+To add default pre-pinned resources:
+
+```text
+├── spec
+│   ├── customization
+|       ├── perspectives
+|           ├── id
+|           ├── ...
+|           ├── pinnedResources
+|               └── group
+                └── version
+                └── resource
+└── ...
+```
+
+Each perspective conforms to the following schema :
+
+- `pinnedResources: operatorv1.PinnedResourceReference[]` , defines a list of pinned resources. The default pinned resources are contributed by the console UI extension plugin.  
+- `group: string` , defines the group of the resource.
+- `version: string` , defines the version of the resource.
+- `resource: string` , defines the type of the resource. It is the plural form of the resource kind in lowercase.
+
+If the `pinnedResources` list is empty, the default pinned resources contributed by the console UI extension plugin will be shown.
+If the `pinnedResources` list is non-empty, only the listed items will be shown if the user hasn't already customized the navigation items.
+If the `group`, `version` or `resource` is incorrect, the resource to be pinned will be ignored.
+
+Note: The Administrator perspective (id="admin") doesn't support pinned resources at the moment and can be excluded via CEL validation.
```

Comment:
What about unrecognized perspectives / perspectives contributed by plugins?

Reply:
So, should we update the CEL to allow adding pinnedResources only to the "dev" perspective?
cc @jerolimov

Reply:
Other perspectives supports pinned resources as well. I think any other then admin should be fine. So we are good here I would expect at the moment.

Reply:
> Hm, I don't see a way to pin resources to the "All Clusters" perspective for ACM/MCE when I test it in the UI. We should double check this.
> 
I have updated this to:
```
Note: The Administrator perspective (id="admin") and "All Clusters" perspective for ACM/MCE (id="acm") doesn't support pinned resources at the moment and can be excluded via CEL validation.
```

> @jerolimov We should also test that we don't have a race condition here:
> 
> https://github.com/openshift/console/blob/master/frontend/packages/console-app/src/components/nav/PerspectiveNav.tsx#L28-L31
> 
> We perform API discovery asynchronously, and the cached resources won't always match the current on initial console load. Also on the first load there will be no cached resources, which could hide links to some resources when a user first uses the console. Really we should react when the models change.

We tested this by clearing the local storage and then refreshed the page and here's a small recording of that, please let us know if there's any other way to test this:

https://user-images.githubusercontent.com/22490998/207379291-5fa26aae-2815-4b43-93bb-c56f2384d7e7.mov